test/watchman_test.exs

```diff
@@ -161,4 +161,41 @@ defmodule WatchmanTest do
     assert len <= max_buffer_size
   end
 
+  describe ".whitelist" do
+    setup do
+      TestUDPServer.wait_for_clean_message_box()
+      TestUDPServer.flush()
+
+      pid = Process.whereis(Watchman.Server)
+      Process.exit(pid, :kill)
+      Watchman.Server.start_link([whitelist: "^external.*"])
```

Comment:
To avoid that race issue you mentioned we can use:
```
Application.put_env(:watchman, :whitelist, "^external.*")
Process.exit(pid, :kill)
```
That way, the server will use the proper config once it is restarted by the supervisor.

Also, this should be added at the end to reset the state for the following test
```
on_exit(fn -> Application.put_env(:watchman, :whitelist, ".*") end)
```

Reply:
This is exactly what I needed!